ons-spark/spark-functions/logistic-regression.md

```diff
@@ -0,0 +1,1660 @@
+
+# Indexing the originofcallcolumn
+callIdx = StringIndexer(inputCol='originofcall',
+                               outputCol='callIndex')
+
+# Indexing the propertycategory column
+propertyIdx = StringIndexer(inputCol='propertycategory',
+                               outputCol='propertyIndex')
+                               
+# Apply indexing to each column one by one
+
+rescue_cat_indexed = serviceIdx.fit(rescue_cat).transform(rescue_cat)
+rescue_cat_indexed = callIdx.fit(rescue_cat_indexed).transform(rescue_cat_indexed)
+rescue_cat_indexed = propertyIdx.fit(rescue_cat_indexed).transform(rescue_cat_indexed)
+
+# Check that this has worked correctly
+rescue_cat_indexed.select('is_cat', 'specialservicetypecategory', 'originofcall',
+                          'propertycategory', 'serviceIndex', 'callIndex', 
+                          'propertyIndex').show(10)
+```
+
+```{code-tab} plaintext Python output
++------+-------------------------------+------------------+-----------------+------------+---------+-------------+
+|is_cat|specialservicetypecategory     |originofcall      |propertycategory |serviceIndex|callIndex|propertyIndex|
++------+-------------------------------+------------------+-----------------+------------+---------+-------------+
+|0     |Other Animal Assistance        |Person (land Line)|Dwelling         |0.0         |1.0      |0.0          |
+|0     |Other Animal Assistance        |Person (land Line)|Outdoor Structure|0.0         |1.0      |3.0          |
+|0     |Animal Rescue From Below Ground|Person (mobile)   |Outdoor Structure|2.0         |0.0      |3.0          |
+|0     |Animal Rescue From Water       |Person (mobile)   |Non Residential  |3.0         |0.0      |2.0          |
+|0     |Other Animal Assistance        |Person (mobile)   |Dwelling         |0.0         |0.0      |0.0          |
+|0     |Other Animal Assistance        |Person (land Line)|Dwelling         |0.0         |1.0      |0.0          |
+|0     |Other Animal Assistance        |Person (land Line)|Outdoor          |0.0         |1.0      |1.0          |
+|0     |Animal Rescue From Water       |Person (mobile)   |Outdoor          |3.0         |0.0      |1.0          |
+|0     |Animal Rescue From Height      |Person (land Line)|Dwelling         |1.0         |1.0      |0.0          |
+|0     |Animal Rescue From Water       |Person (mobile)   |Outdoor          |3.0         |0.0      |1.0          |
++------+-------------------------------+------------------+-----------------+------------+---------+-------------+
+```
+````
+Next we use `OneHotEncoderEstimator` to perform one-hot encoding. This converts each of the categories we have applied the `StringIndexer` to into a vector that indicates which category the record belongs to. 
+
+````{tabs}
+```{code-tab} py
+# Apply OneHotEncoderEstimator to each categorical column simultaneously
+encoder = OneHotEncoderEstimator(inputCols = ['serviceIndex', 'callIndex', 'propertyIndex'], 
```

Comment:
Needs updating to use OneHotEncoder as OneHotEncoderEstimator has been deprecated